src/Castle.Core/DynamicProxy/Generators/BaseProxyGenerator.cs

```diff
@@ -396,9 +396,9 @@ protected Type ObtainProxyType(CacheKey cacheKey, Func<string, INamingScope, Typ
 			}
 
 			// This is to avoid generating duplicate types under heavy multithreaded load.
```

Comment:
nit, you could probably move the comment 'This is to avoid generating duplicate types under heavy multithreaded load'  to just after 'See if an earlier lock holder populated the cache.'

Reply:
Actually, I think the comment `// See if an earlier lock holder populated the cache.` should be moved up to the first cache check.